FirebaseInstallations/Source/Library/InstallationsIDController/FIRCurrentDateProvider.m

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FirebaseInstallations/Source/Library/InstallationsIDController/FIRCurrentDateProvider.h"
+
+FIRCurrentDateProvider FIRRealCurrentDateProvider(void) {
```

Comment:
Is the date provider only for testability? Or is there any other reason to have something different than `[NSDate date]`?

Reply:
Yes, it's for testability only. `[NSDate date]` and similar methods are so widely used by SDKs and system that mocking it often leads to weird flaky tests failures. Ideally we should never mock `NSDate` methods. Let me add explanation to the comments as well.